### PR TITLE
read segments from JWT (DEV-7295)

### PR DESCRIPTION
**Pull request by peter-horvath**

### CODE REVIEW

File: test/JwtTokenAuthenticatorTest.php [hunk 1]
```diff
@@ -229,6 +247,14 @@ public function givenTokenWithoutGroupId_whenVerifyingToken_returnTrue()
     /**
      * @test
      */
+    public function givenTokenWithoutSegments_whenVerifyingToken_returnTrue()
```
Comment (luos) [inline test/JwtTokenAuthenticatorTest.php:250]: Why is a token without segments valid?

Comment (luos) [inline test/JwtTokenAuthenticatorTest.php:250]: Ah, I think I see it.

**Comment by luos** [comment]
Looks OK :100:

File: test/JwtTokenAuthenticatorTest.php (hunk 1 above)
Comment (peter-horvath) [inline test/JwtTokenAuthenticatorTest.php:250]: I don't want to revoke access from users with tokens issued without segments by a previous version of the code. I think right now they would be locked out from preview until they delete their cookies or until their token expires. Most users won't have segments for now anyway.

**Pull request merged**